credentials/credentials_test.go

```diff
@@ -12,75 +12,254 @@ import (
+		}
+	}
+
+	return nil, fmt.Errorf("Parameter not found")
 }
 
-func TestGetNewRelicLicenseKeyConfigValue(t *testing.T) {
-	licenseKey := "test_value"
+func TestIsParameterConfigured(t *testing.T) {
 	ctx := context.Background()
-	resultKey, err := GetNewRelicLicenseKey(ctx, &config.Configuration{
-		LicenseKey: licenseKey,
+	assert.False(t, IsParameterConfigured(ctx, &config.Configuration{}))
+
+	OverrideSSM(mockSSM{
+		validParameters: []string{"testParameterName"},
 	})
+	assert.True(t, IsParameterConfigured(ctx, &config.Configuration{
+		LicenseKeyParameterName: "testParameterName",
+	}))
+
+	OverrideSSM(mockSSM{})
+	assert.False(t, IsParameterConfigured(ctx, &config.Configuration{
+		LicenseKeyParameterName: "testParameterName",
+	}))
+}
+
+func TestGetNewRelicLicenseKey(t *testing.T) {
+	table := []struct {
+		Name string
+
+		Conf           config.Configuration
+		Environment    map[string]string
+		SecretsManager secretsmanageriface.SecretsManagerAPI
+		SSM            ssmiface.SSMAPI
+
+		ExpectedKey string
+		ExpectedErr string
+	}{
+		{
+			Name: "uses config license key when present",
+			Conf: config.Configuration{
+				LicenseKey: "in_config",
+			},
+
+			ExpectedKey: "in_config",
+		},
+		{
+			Name: "loads license key from secret when configured",
+			Conf: config.Configuration{
+				LicenseKeySecretId: "testSecretName",
+			},
+			SecretsManager: mockSecretManager{
+				validSecrets: []string{"testSecretName"},
+			},
+
+			ExpectedKey: mockSecretManagerKeyValue,
+		},
+		{
+			Name: "loads license key from parameter when configured",
+			Conf: config.Configuration{
+				LicenseKeyParameterName: "testParameterName",
+			},
+			SSM: mockSSM{
+				validParameters: []string{"testParameterName"},
+			},
+
+			ExpectedKey: mockParameterStoreKeyValue,
+		},
+		{
+			Name: "loads license key from environment variable if not configured",
+			Conf: config.Configuration{},
+			Environment: map[string]string{
+				"NEW_RELIC_LICENSE_KEY": "12345",
+			},
+
+			ExpectedKey: "12345",
+		},
+		{
+			Name: "loads license key from environment variable if configured secret is unavailable",
+			Conf: config.Configuration{
+				LicenseKeySecretId: "testSecretName",
+			},
+			SecretsManager: mockSecretManager{},
+			Environment: map[string]string{
+				"NEW_RELIC_LICENSE_KEY": "12345",
+			},
+
+			ExpectedKey: "12345",
+		},
+		{
+			Name: "loads license key from environment variable if configured parameter is unavailable",
+			Conf: config.Configuration{
+				LicenseKeyParameterName: "testParameterName",
+			},
+			SSM: mockSSM{},
+			Environment: map[string]string{
+				"NEW_RELIC_LICENSE_KEY": "12345",
+			},
+
+			ExpectedKey: "12345",
+		},
+		{
+			Name: "returns error if configured secret is unavailable",
+			Conf: config.Configuration{
+				LicenseKeySecretId: "testSecretName",
+			},
+			SecretsManager: mockSecretManager{},
+
+			ExpectedErr: "Secret not found",
+		},
+		{
+			Name: "returns error if configured parameter is unavailable",
```

Comment:
NP Maybe add "and license key environment variable is not provided" to the names?